main/main.cpp

```diff
@@ -2325,6 +2325,8 @@ bool Main::iteration() {
 	}
 #endif
 
+	ProjectSettings::get_singleton()->decrement_dirty_this_frame();
```

Comment:
I'm not super happy about having something like this injected in the utmost core `Main::iteration`. Feels hacky, and not just because it has "dirty" in the name :)

Reply:
Ah yes fair point, I agree. :+1: There's probably a better place for it I'll try and have a hunt (any suggestions welcome).